### ISSUE TITLE
Customizing KeyboardIcon and PenIcon for keyboard view toggle

### ISSUE THREAD

**Issue by Philipp91** [mui-org/material-ui-pickers#1693]
The icons are currently [hard-coded](https://github.com/mui-org/material-ui-pickers/blob/c413b02149ecad16c928fd043967509b1f5f42f3/lib/src/_shared/PickerToolbar.tsx#L104-L106).

There is already a `keyboardIcon` prop, but that one controls the icon in the input adornment. I don't really understand why that prop has "keyboard" in its name. It does not show a keyboard on the icon. And one would click that button precisely when _not_ using the keyboard but rather the mouse to input a date. I'd call that one `openIcon` (by function) or `calendarIcon` (by icon content).

**Comment by dmtrKovalenko** [comment]
Yeah, it is reaaly legacy name. Should be nice to change in scope of `v4-alpha` breaking changes

**Comment by dmtrKovalenko** [comment]
Would you like to open PR?

**Comment by Philipp91** [comment]
Well it's not just about renaming the existing prop, but also about exposing new props for those two hard-coded icons.

**Comment by dmtrKovalenko** [comment]
Yeah so I propose these changes:

1. Rename `keyboardIcon` =>  `calendarIcon` 
2. Use this icon also for rendering mobile keyboard input view 
3. Add new prop `editIcon` for the mobile keyboard input view

**Comment by Philipp91** [comment]
Sure, I can take a stab at this. I can't assign myself in GitHub, but stay tuned.

**Comment by Philipp91** [comment]
> Yeah so I propose these changes:
> 
> 1. Rename `keyboardIcon` =>  `calendarIcon`

Naming the prop after the icon content doesn't play well with [this usage](https://github.com/mui-org/material-ui-pickers/blob/341171f5ae4e600f8d8efcfebf3837adbbb293cc/lib/src/TimePicker/TimePicker.tsx#L41), where we'd end up with `calendarIcon: <ClockIcon/>` even those two are sort-of opposites and shouldn't be equated like that.

So perhaps `openIcon` or something else in the direction of the icon's functionality (adornmentIcon, buttonIcon) would be better?

**Comment by dmtrKovalenko** [comment]
Hmm really, `openPickerIcon` should be nice?

**Comment by Philipp91** [comment]
> 2. Use this icon also for rendering mobile keyboard input view
> 3. Add new prop editIcon for the mobile keyboard input view

Just like for the other one, I don't think we should name the props after the icon content. How about `mobileKeyboardViewIcon` (that's the pen that switches to the textbox/keyboard view) and `mobileDefaultViewIcon` (that's the calendar icon that switches back to the regular mobile view)?
